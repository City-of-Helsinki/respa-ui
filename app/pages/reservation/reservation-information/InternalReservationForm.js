/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import Form from 'react-bootstrap/lib/Form';
import Row from 'react-bootstrap/lib/Row';
import Col from 'react-bootstrap/lib/Col';

import injectT from '../../../i18n/injectT';
import FormTypes from '../../../constants/FormTypes';

export function validate() {
  const errors = {};
  return errors;
}

class UnconnectedInternalReservationForm extends Component {
  state = {
    internalReservationDefaultChecked: true
  }

  render() {
    /**
     * TODO Add FI/SV/EN translations!
     */
    const premiseStaffOnly = 'Premise staff only';
    const internalReservation = 'Internal reservation';
    const internalReservationDescription = 'The Department\'s own event. Uncheck this box to make an external reservation on behalf of the user.';
    const markAsClosed = 'Mark as closed';
    const markAsClosedDescription = 'Check if you are closing the resource by reserving.';
    const comment = 'Comment';
    return (
      <div className="app-ReservationDetails">
        <Form className="reservation-form" horizontal noValidate>
          <h2 className="app-ReservationPage__title">{premiseStaffOnly}</h2>
          <Row>
            <Col md={1}>
              <Field
                checked={this.state.internalReservationDefaultChecked}
                component="input"
                name="internalReservation"
                onChange={() => this.setState(prevState => ({ internalReservationDefaultChecked: !prevState.internalReservationDefaultChecked }))}
                type="checkbox"
              />
            </Col>
            <Col md={11}>
              <span className="app-ReservationDetails__value">
                {internalReservation}
                <br />
                {internalReservationDescription}
              </span>
            </Col>
          </Row>
          <Row>
            <Col md={1}>
              <Field
                component="input"
                name="markAsClosed"
                type="checkbox"
              />
            </Col>
            <Col md={11}>
              <span className="app-ReservationDetails__value">
                {markAsClosed}
                <br />
                {markAsClosedDescription}
              </span>
            </Col>
          </Row>
          <Row>
            <Col md={12}>
              <div className="app-ReservationPage__formfield">
                <label>
                  {comment}
                  <Field
                    component="textarea"
                    name="internalReservationComments"
                    rows={5}
                  />
                </label>
              </div>
            </Col>
          </Row>
        </Form>
      </div>
    );
  }
}

UnconnectedInternalReservationForm.propTypes = {};
UnconnectedInternalReservationForm = injectT(UnconnectedInternalReservationForm);  // eslint-disable-line

export { UnconnectedInternalReservationForm };
export default injectT(reduxForm({
  form: FormTypes.RESERVATION,
  validate
})(UnconnectedInternalReservationForm));
