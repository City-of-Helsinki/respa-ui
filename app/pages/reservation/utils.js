import client from '../../../src/common/api/client';

/**
 * Get reservation detail
 *
 * @param {Object} reservationId Filters
 * @returns {Promise}
 */
export const getReservationById = (reservationId) => {
  return client.get(`reservation/${reservationId}`);
};

/**
 * Get resource detail
 *
 * @param {Object} filters Filters
 * @returns {Promise}
 */
export const getResourceById = (resourceId) => {
  return client.get(`resource/${resourceId}`);
};
