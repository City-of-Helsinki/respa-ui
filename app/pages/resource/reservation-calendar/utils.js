import filter from 'lodash/filter';
import maxBy from 'lodash/maxBy';
import minBy from 'lodash/minBy';
import some from 'lodash/some';
import moment from 'moment';

import constants from '../../../constants/AppConstants';

function getBeginOfSelection(selected) {
  return minBy(selected, 'begin');
}

function getEndOfSelection(selected) {
  return maxBy(selected, 'begin');
}

function getNextDayFromDate(date) {
  return date
    ? moment(date)
      .add(1, 'days')
      .format(constants.DATE_FORMAT)
    : null;
}

function getNextWeeksDays(date) {
  const selectedWeekDay = moment(date).day();
  if (selectedWeekDay === 0) {
    return 2;
  } if (selectedWeekDay === 6) {
    return 1;
  }
  return 0;
}

function getSecondDayFromDate(date) {
  return date
    ? moment(date)
      .add(2, 'days')
      .format(constants.DATE_FORMAT)
    : null;
}

function isInsideOpeningHours(slot, openingHours) {
  const date = moment(slot.start).format(constants.DATE_FORMAT);
  const slotOpeningHours = filter(openingHours, { date });
  return some(
    slotOpeningHours,
    opening => moment(opening.opens).isSameOrBefore(slot.start)
      && moment(slot.end).isSameOrBefore(opening.closes)
  );
}

function isSlotAfterSelected(slot, selected) {
  if (slot.editing) {
    return false;
  }
  const firstSelected = getBeginOfSelection(selected);
  return firstSelected.begin <= slot.start;
}

function isSlotSelectable(slot, selected, resource, lastSelectableFound, isAdmin) {
  if (!slot || !selected || !selected.length || !resource) {
    return true;
  }
  if ((!slot.editing && slot.reserved) || lastSelectableFound) {
    return false;
  }
  const firstSelected = getBeginOfSelection(selected);
  if (!isAdmin && resource.maxPeriod) {
    const durationParts = resource.maxPeriod.split(':');
    // eslint-disable-next-line no-mixed-operators
    const maxPeriodMinutes = 60 * durationParts[0] + durationParts[1];
    const maxEndDate = moment(firstSelected.begin);
    maxEndDate.minutes(maxEndDate.minutes() + maxPeriodMinutes);
    if (moment(slot.start).isSameOrAfter(maxEndDate)) {
      return false;
    }
  }
  const firstSelectedDate = moment(firstSelected.begin);
  const slotStartDate = moment(slot.start);
  return (
    moment(firstSelectedDate).isSameOrBefore(slotStartDate) && moment(firstSelectedDate).isSame(slotStartDate, 'date')
  );
}

function isSlotSelected(slot, selected) {
  if (!slot || !selected || !selected.length) {
    return false;
  }
  const firstSelected = getBeginOfSelection(selected);
  const lastSelected = getEndOfSelection(selected);
  return firstSelected.begin <= slot.start && lastSelected.end >= slot.end;
}

function isFirstSelected(slot, selected) {
  if (!slot || !selected || !selected.length) {
    return false;
  }
  const firstSelected = getBeginOfSelection(selected);
  return firstSelected.begin === slot.start;
}

function isHighlighted(slot, selected, hovered) {
  if (!slot || !selected || !hovered || !selected.length) {
    return false;
  }
  const firstSelected = getBeginOfSelection(selected);
  const firstSelectedDate = moment(firstSelected.begin);
  const slotStartDate = moment(slot.start);
  const hoveredDate = moment(hovered.start);
  return (
    slotStartDate > firstSelectedDate
    && slotStartDate < hoveredDate
    && firstSelectedDate.isSame(slotStartDate, 'date')
  );
}
/**
 * Check if current slot will over minPeriod
 * by adding minPeriod to slot start time
 * and compare with last slot end time
 *
 * @param {Object} slot
 * @param {Object} lastSlot
 * @param {String | undefined} minPeriod: minPeriod limit, usuall HH:MM:SS
 * @returns
 */
function isUnderMinPeriod(slot, lastSlot, minPeriod) {
  if (!slot || !lastSlot) {
    return false;
  }

  if (!slot.end || !lastSlot.end) {
    return false;
  }
  if (minPeriod) {
    const minPeriodInMinutes = moment.duration(minPeriod).asMinutes();
    return moment(slot.start).add(minPeriodInMinutes, 'minutes') > (moment(lastSlot.end));
  }

  return false;
}

export default {
  getNextDayFromDate,
  getNextWeeksDays,
  getSecondDayFromDate,
  isHighlighted,
  isInsideOpeningHours,
  isSlotAfterSelected,
  isSlotSelectable,
  isSlotSelected,
  isFirstSelected,
  isUnderMinPeriod,
};
