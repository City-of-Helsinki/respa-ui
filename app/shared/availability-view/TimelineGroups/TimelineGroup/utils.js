import some from 'lodash/some';
import moment from 'moment';

import { slotSize, slotWidth, slotMargin } from 'shared/availability-view';

function getTimeSlotWidth({ startTime, endTime } = {}) {
  const diff = endTime ? endTime.diff(startTime, 'minutes') : slotSize;
  const slots = Math.floor(diff / slotSize);

  return (slotWidth * slots) - slotMargin;
}

function getTimelineItems(date, reservations, resourceId) {
  const items = [];
  let reservationPointer = 0;
  let timePointer = date.clone().startOf('day');
  const end = date.clone().endOf('day');
  while (timePointer.isBefore(end)) {
    const reservation = reservations && reservations[reservationPointer];
    const isSlotReservation = reservation && timePointer.isSame(reservation.begin);
    if (isSlotReservation) {
      items.push({
        key: String(items.length),
        type: 'reservation',
        data: reservation,
      });
      timePointer = moment(reservation.end);
      reservationPointer += 1;
    } else {
      items.push({
        key: String(items.length),
        type: 'reservation-slot',
        data: {
          begin: timePointer.format(),
          end: timePointer.clone().add(slotSize, 'minutes').format(),
          resourceId,
        },
      });
      timePointer.add(slotSize, 'minutes');
    }
  }
  return items;
}

function isInsideAvailableHours(item, availableHours) {
  return some(availableHours, availability => (
    availability.starts <= item.data.begin && item.data.end <= availability.ends
  ));
}

function markItemSelectable(item, isSelectable, availableHours) {
  const isInFuture = moment().isSameOrBefore(item.data.end);
  const selectable = isInFuture && (
    isSelectable && availableHours ?
    isInsideAvailableHours(item, availableHours) :
    isSelectable
  );
  return { ...item, data: { ...item.data, isSelectable: selectable } };
}

function markItemsSelectable(items, isSelectable, availableHours) {
  return items.map((item) => {
    if (item.type === 'reservation') return item;
    return markItemSelectable(item, isSelectable, availableHours);
  });
}

function addSelectionData(selection, resource, items) {
  if (!selection) {
    return markItemsSelectable(items, true, resource.availableHours);
  } else if (selection.resourceId !== resource.id) {
    return markItemsSelectable(items, false);
  }
  let lastSelectableFound = false;
  return items.map((item) => {
    if (lastSelectableFound || item.data.begin < selection.begin) {
      if (item.type === 'reservation') return item;
      return markItemSelectable(item, false);
    }
    if (item.type === 'reservation') {
      lastSelectableFound = true;
      return item;
    }
    return markItemSelectable(item, true, resource.availableHours);
  });
}

export default {
  addSelectionData,
  getTimelineItems,
  getTimeSlotWidth,
};
