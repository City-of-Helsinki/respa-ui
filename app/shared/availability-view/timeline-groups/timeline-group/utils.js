import some from 'lodash/some';
import moment from 'moment';

import {
  slotSize,
  slotWidth,
  slotMargin,
  slotFullDayWidth, slotMinuteWidth,
} from '../../../../constants/SlotConstants';
import { reservationLengthType } from '../../../../../src/domain/resource/constants';

function getMultidayPeriod(resource, date) {
  if (!resource.periods) {
    return undefined;
  }
  return resource.periods.find(
    period => moment(period.start).isSameOrBefore(moment(date))
      && moment(period.end).isSameOrAfter(moment(date))
      && (period.reservationLengthType === reservationLengthType.WHOLE_DAY
        || period.reservationLengthType === reservationLengthType.OVERNIGHT),
  );
}

function getWithinDayTimeSlotWidth(startTime, endTime) {
  const diff = endTime ? moment(endTime).diff(moment(startTime), 'minutes') : slotSize;
  const slots = Math.round(diff / slotSize);

  return slotWidth * slots - slotMargin;
}

function getMultidayTimeSlotWidth(startTime, endTime, date) {
  const isStartTimeBeforeDate = moment(startTime).isBefore(moment(date), 'day');
  const isEndTimeAfterDate = moment(endTime).isAfter(moment(date), 'day');
  const isWholeDay = isStartTimeBeforeDate && isEndTimeAfterDate;

  if (isWholeDay) {
    return slotFullDayWidth;
  }

  if (isStartTimeBeforeDate) {
    return moment(endTime).diff(moment(date).startOf('day'), 'minutes') * slotMinuteWidth;
  }

  if (isEndTimeAfterDate) {
    return moment(date).endOf('day').diff(moment(startTime), 'minutes') * slotMinuteWidth;
  }

  // Neither start nor end time is outside given date, get the within day slot width.
  return getWithinDayTimeSlotWidth(startTime, endTime);
}

function getTimeSlotWidth({ startTime, endTime, date } = {}) {
  if (!startTime || !endTime || !date) {
    return getWithinDayTimeSlotWidth(startTime, endTime);
  }

  const isWithinDay = moment(startTime).isSame(moment(date), 'day')
    && moment(endTime).isSame(moment(date), 'day');
  if (isWithinDay) {
    return getWithinDayTimeSlotWidth(startTime, endTime);
  }

  return getMultidayTimeSlotWidth(startTime, endTime, date);
}

function getMultidayTimelineItems(multidayPeriod, date, resource) {
  const resourceId = resource.id;
  const reservations = Array.isArray(resource.reservations)
    ? resource.reservations.filter(
      reservation => reservation.state !== 'cancelled' && reservation.state !== 'denied',
    )
    : [];
  const endingTime = {
    hours: moment
      .duration(multidayPeriod.multidaySettings.checkOutTime)
      .hours(),
    min: moment
      .duration(multidayPeriod.multidaySettings.checkOutTime)
      .minutes(),
  };
  const startingTime = {
    hours: moment.duration(multidayPeriod.multidaySettings.checkInTime).hours(),
    min: moment.duration(multidayPeriod.multidaySettings.checkInTime).minutes(),
  };

  const isWholeDayPeriod = multidayPeriod.reservationLengthType === reservationLengthType.WHOLE_DAY;
  const isOvernightPeriod = multidayPeriod.reservationLengthType === reservationLengthType.OVERNIGHT;

  const wholeDayReservation = reservations.find(
    reservation => moment(reservation.begin).isBefore(moment(date), 'day')
      && moment(reservation.end).isAfter(moment(date), 'day'),
  );
  const reservationEndingOnDate = reservations.find(
    reservation => moment(reservation.begin).isBefore(moment(date), 'day')
      && moment(reservation.end).isSame(moment(date), 'day'),
  );
  const reservationStartingOnDate = reservations.find(
    reservation => moment(reservation.end).isAfter(moment(date), 'day')
      && moment(reservation.begin).isSame(moment(date), 'day'),
  );

  const isWholeDayReserved = !!wholeDayReservation;
  const hasPartialDayReserved = reservationEndingOnDate || reservationStartingOnDate;

  // We want to populate the timeline as follows:

  // 1. For any day with no reservations, draw one single clickable block.
  if (!hasPartialDayReserved && !isWholeDayReserved) {
    return [
      {
        key: '1',
        type: 'multiday-reservation-slot',
        data: {
          begin: moment(date).startOf('day').format(),
          end: moment(date).endOf('day').format(),
          resourceId,
          isSelectable: true,
          reservationPeriod: multidayPeriod,
        },
      },
    ];
  }

  // 2. For days that are completely covered by a reservation, draw one single reservation block.
  if (isWholeDayReserved) {
    return [
      {
        key: '1',
        type: 'reservation',
        data: wholeDayReservation,
      },
    ];
  }
  // 3. If reservation only covers day partially, but period type is WHOLE_DAY
  if (isWholeDayPeriod && reservationEndingOnDate) {
    return [
      {
        key: '1',
        type: 'reservation',
        data: reservationEndingOnDate,
      },
      {
        key: '2',
        type: 'multiday-reservation-slot',
        data: {
          begin: moment(date)
            .startOf('day')
            .add(endingTime.hours, 'hours')
            .add(endingTime.min, 'minutes')
            .format(),
          end: moment(date).endOf('day').format(),
          resourceId,
          isSelectable: false,
          reservationPeriod: multidayPeriod,
        },
      },
    ];
  }
  if (isWholeDayPeriod && reservationStartingOnDate) {
    return [
      {
        key: '1',
        type: 'multiday-reservation-slot',
        data: {
          begin: moment(date).startOf('day').format(),
          end: moment(date)
            .startOf('day')
            .add(startingTime.hours, 'hours')
            .add(startingTime.min, 'minutes')
            .format(),
          resourceId,
          isSelectable: false,
          reservationPeriod: multidayPeriod,
        },
      },
      {
        key: '2',
        type: 'reservation',
        data: reservationStartingOnDate,
      },
    ];
  }

  // 3. If period type is OVERNIGHT and reservation doesn't cover the whole day:
  if (isOvernightPeriod && hasPartialDayReserved && !isWholeDayReserved) {
    // 3a. Has a single reservation from the beginning of the day to checkout time: reservation block,
    // followed by blank, clickable block from check-in to the end of the day.
    if (!reservationStartingOnDate && reservationEndingOnDate) {
      return [
        {
          key: '1',
          type: 'reservation',
          data: reservationEndingOnDate,
        },
        {
          key: '2',
          type: 'multiday-reservation-slot',
          data: {
            begin: moment(date)
              .startOf('day')
              .add(endingTime.hours, 'hours')
              .add(endingTime.min, 'minutes')
              .format(),
            end: moment(date)
              .startOf('day')
              .add(startingTime.hours, 'hours')
              .add(startingTime.min, 'minutes')
              .format(),
            resourceId,
            isSelectable: false,
            reservationPeriod: multidayPeriod,
          },
        },
        {
          key: '3',
          type: 'multiday-reservation-slot',
          data: {
            begin: moment(date)
              .startOf('day')
              .add(startingTime.hours, 'hours')
              .add(startingTime.min, 'minutes')
              .format(),
            end: moment(date).endOf('day').format(),
            resourceId,
            isSelectable: true,
            reservationPeriod: multidayPeriod,
          },
        },
      ];
    }

    // 3b. Has two reservations, one until checkout and one from check-in until the end of the day:
    // two reservation blocks with blank in between.
    if (reservationStartingOnDate && reservationEndingOnDate) {
      return [
        {
          key: '1',
          type: 'reservation',
          data: reservationEndingOnDate,
        },
        {
          key: '2',
          type: 'multiday-reservation-slot',
          data: {
            begin: moment(date)
              .startOf('day')
              .add(endingTime.hours, 'hours')
              .add(endingTime.min, 'minutes')
              .format(),
            end: moment(date)
              .startOf('day')
              .add(startingTime.hours, 'hours')
              .add(startingTime.min, 'minutes')
              .format(),
            resourceId,
            isSelectable: false,
            reservationPeriod: multidayPeriod,
          },
        },
        {
          key: '3',
          type: 'reservation',
          data: reservationStartingOnDate,
        },
      ];
    }

    // 3c. Has a single reservation from check-in until end of the day: fill with blank until check-in,
    // reservation block for the rest.
    if (reservationStartingOnDate) {
      return [
        {
          key: '1',
          type: 'multiday-reservation-slot',
          data: {
            begin: moment(date).startOf('day').format(),
            end: moment(date)
              .startOf('day')
              .add(startingTime.hours, 'hours')
              .add(startingTime.min, 'minutes')
              .format(),
            resourceId,
            isSelectable: false,
            reservationPeriod: multidayPeriod,
          },
        },
        {
          key: '2',
          type: 'reservation',
          data: reservationStartingOnDate,
        },
      ];
    }
  }

  // All cases covered, this return is here to keep the linter happy.
  return [];
}

function isInsideOpeningHours(item, openingHours) {
  return some(
    openingHours,
    opening => opening.opens <= item.data.begin && item.data.end <= opening.closes,
  );
}

function markItemSelectable(item, isSelectable, openingHours) {
  const selectable = isSelectable
    && moment().isSameOrBefore(item.data.end)
    && (!openingHours || isInsideOpeningHours(item, openingHours));
  return { ...item, data: { ...item.data, isSelectable: selectable } };
}

function markItemsSelectable(items, isSelectable, openingHours) {
  return items.map((item) => {
    if (item.type === 'reservation') return item;
    return markItemSelectable(item, isSelectable, openingHours);
  });
}

function addSelectionData(selection, resource, items) {
  if (!selection) {
    return markItemsSelectable(items, true, resource.openingHours);
  }
  if (selection.resourceId !== resource.id) {
    // isSelectable is false by default, so nothing needs to be done.
    // This is a pretty important performance optimization when there are tons of
    // resources in the AvailabilityView and the selection is in a state where the
    // first click has been done but the second (end time) hasn't. Without this
    // optimization we'd be calling markItemSelectable for every slot in every
    // resource when the user hovers to another slot.
    return items;
  }
  let lastSelectableFound = false;
  return items.map((item) => {
    if (lastSelectableFound || item.data.begin < selection.begin) {
      if (item.type === 'reservation') return item;
      // isSelectable is false by default.
      return item;
    }
    if (item.type === 'reservation') {
      lastSelectableFound = true;
      return item;
    }
    return markItemSelectable(item, true, resource.openingHours);
  });
}

function getTimelineItems(resource, date, reservations, resourceId, selection) {
  const multidayPeriod = getMultidayPeriod(resource, date);
  if (multidayPeriod) {
    return getMultidayTimelineItems(multidayPeriod, date, resource);
  }

  const items = [];
  let reservationPointer = 0;
  let timePointer = date.clone().startOf('day');
  const end = date.clone().endOf('day');
  while (timePointer.isBefore(end)) {
    const reservation = reservations && reservations[reservationPointer];
    const isSlotReservation = reservation && timePointer.isSame(reservation.begin);
    if (isSlotReservation) {
      items.push({
        key: String(items.length),
        type: 'reservation',
        data: reservation,
      });
      timePointer = moment(reservation.end);
      reservationPointer += 1;
    } else {
      items.push({
        key: String(items.length),
        type: 'reservation-slot',
        data: {
          begin: timePointer.format(),
          end: timePointer.clone().add(slotSize, 'minutes').format(),
          resourceId,
          // isSelectable: false by default to improve selector performance by allowing
          // addSelectionData to make some assumptions.
          isSelectable: false,
        },
      });
      timePointer.add(slotSize, 'minutes');
    }
  }
  return addSelectionData(selection, resource, items);
}

export default {
  getTimelineItems,
  getTimeSlotWidth,
};
