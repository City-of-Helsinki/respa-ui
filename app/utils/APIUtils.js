import { camelizeKeys } from 'humps';
import { normalize } from 'normalizr';

export default {
  createTransformFunction,
};

function createTransformFunction(schema) {
  return (json) => {
    const camelizedJson = camelizeKeys(json);
    if (schema) {
      return normalize(camelizedJson, schema);
    }
    return camelizedJson;
  };
}
