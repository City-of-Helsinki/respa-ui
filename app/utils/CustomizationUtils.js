import { CUSTOMIZATIONS } from 'constants/AppConstants';

function getCurrentCustomization() {
  const host = window.location.host;
  return CUSTOMIZATIONS[host] || null;
}

export default {
  getCurrentCustomization,
};
