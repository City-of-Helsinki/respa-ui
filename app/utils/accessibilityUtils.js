import { getFromLocalStorage, storeInLocalStorage } from './localStorageUtils';

const A11Y_LOCAL_STORAGE_KEY = 'varaamo_selectedViewpoints';

export const getSelA11yPref = () => getFromLocalStorage(A11Y_LOCAL_STORAGE_KEY) || [];

export const setSelA11yPref = selectedViewpoints => storeInLocalStorage(A11Y_LOCAL_STORAGE_KEY, selectedViewpoints);
