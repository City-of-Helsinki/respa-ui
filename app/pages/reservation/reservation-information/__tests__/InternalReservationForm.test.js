import React from 'react';
import toJSON from 'enzyme-to-json';

import { shallowWithIntl } from '../../../../utils/testUtils';
import { UnconnectedInternalReservationForm as InternalReservationForm } from '../InternalReservationForm';

describe('pages/reservation/reservation-information/InternalReservationForm', () => {
  describe('validation', () => {
    const input = shallowWithIntl(<InternalReservationForm />).find('#internalReservationChecked');
    it('has input#internalReservationChecked', () => {
      expect(input).toHaveLength(1);
    });
    it('is checked', () => {
      expect(input.props().checked).toEqual(true);
    });
  });
  describe('rendering', () => {
    it('renders correctly', () => {
      const wrapper = shallowWithIntl(<InternalReservationForm />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
