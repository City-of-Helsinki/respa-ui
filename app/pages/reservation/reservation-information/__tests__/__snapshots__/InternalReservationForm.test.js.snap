// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/reservation/reservation-information/InternalReservationForm rendering renders correctly 1`] = `
<div
  className="app-ReservationDetails"
>
  <Form
    bsClass="form"
    className="reservation-form"
    componentClass="form"
    horizontal={true}
    inline={false}
    noValidate={true}
  >
    <h2
      className="app-ReservationPage__title"
    >
      ReservationForm.premiseStaffOnly
    </h2>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={1}
      >
        <label
          className="custom-checkbox-container"
        >
          <Field
            checked={true}
            component="input"
            id="internalReservationChecked"
            name="internalReservation"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="custom-checkmark"
          />
        </label>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={11}
      >
        <span
          className="app-ReservationDetails__value"
        >
          ReservationForm.internalReservation
          <br />
          ReservationForm.internalReservationDescription
        </span>
      </Col>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={1}
      >
        <label
          className="custom-checkbox-container"
        >
          <Field
            checked={false}
            component="input"
            name="markAsClosed"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="custom-checkmark"
          />
        </label>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={11}
      >
        <span
          className="app-ReservationDetails__value"
        >
          ReservationForm.markAsClosed
          <br />
          ReservationForm.markAsClosedDescription
        </span>
      </Col>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={12}
      >
        <div
          className="app-ReservationPage__formfield"
        >
          <label>
            common.comments
            <Field
              component="textarea"
              maxLength={1500}
              name="internalReservationComments"
              rows={5}
            />
          </label>
        </div>
      </Col>
    </Row>
  </Form>
</div>
`;
